internal/users/idgenerator.go

```diff
@@ -133,15 +134,15 @@ func getIDCandidate(minID, maxID uint32, usedIDs []uint32) (uint32, error) {
 	}
 
 	// Try IDs above the highest used
-	for id := highestUsed + 1; id <= maxID; id++ {
+	for id := highestUsed + 1; id <= maxID && highestUsed < math.MaxUint32; id++ {
```

Comment:
This check makes no sense. `highestUsed` is a `uint32`, so `highestUsed < math.MaxUint32` is always true. I think the `id <= maxID` is enough here, I don't see how this loop can cause an integer overflow.

Reply:
> Ah, unless `maxID == math.MaxUint32`, then `id` can overflow to `0`

Exactly...

Reply:
> I still think that highestUsed < math.MaxUint32 is not the check we want here, we should check the loop variable, right?

before you spend time fixing this, please take a look at my proposal in #993 (comment)

Reply:
Yeah, I saw, but again it was already fine even before when mixing the two loops, but still better the new proposal